tests/service/privacy_request/request_runner_service_test.py

```diff
@@ -99,6 +99,22 @@ def test_start_processing_doesnt_overwrite_started_processing_at(
     assert privacy_request.started_processing_at == before
 
 
+def test_halts_proceeding_if_cancelled(
+    db: Session,
+    privacy_request_status_canceled: PrivacyRequest,
+    run_privacy_request_task,
+) -> None:
+    assert privacy_request_status_canceled.status == PrivacyRequestStatus.canceled
+    run_privacy_request_task.delay(privacy_request_status_canceled.id).get(
+        timeout=PRIVACY_REQUEST_TASK_TIMEOUT
+    )
+    _sessionmaker = get_db_session()
+    db = _sessionmaker()
```

Comment:
I'd love to come up with a util to refresh the initial session passed in from fixtures to avoid needing to do this

Reply:
i switched it to refreshing the session, I use that in other places, I just assumed there was some other reason we couldn't do that with these type of tests.